libvast/src/system/index.cpp

```diff
@@ -657,21 +660,15 @@ index(index_actor::stateful_pointer<index_state> self,
       }
     },
     [self](caf::unit_t&, const caf::error& err) {
-      // During "normal" shutdown, the node will send an exit message to
-      // the importer which then cuts the stream to the index, and the
-      // index exits afterwards.
+      VAST_DEBUG("index finalized streaming {}", err);
       // We get an 'unreachable' error when the stream becomes unreachable
-      // during actor destruction; in this case we can't use `self->state`
-      // anymore since it will already be destroyed.
-      VAST_DEBUG("index finalized streaming with error {}", render(err));
+      // because the actor was destroyed; in this case the state was already
+      // destroyed during `local_actor::on_exit()`.
       if (err && err != caf::exit_reason::unreachable) {
-        if (err != caf::exit_reason::user_shutdown)
-          VAST_ERROR("{} got a stream error: {}", self, render(err));
-        else
-          VAST_DEBUG("{} got a user shutdown error: {}", self, render(err));
+        VAST_ERROR("{} aborts with error: {}", self, err);
         // We can shutdown now because we only get a single stream from the
         // importer.
-        self->send_exit(self, err);
+        self->quit(err);
```

Comment:
Does `quit()` still call the exit handler?

Reply:
Ah, that was left in by mistake. Will remove.